### PR TITLE
Cleanup + helpers

### PR DESCRIPTION
**Pull request by subbuss**
Once I have Parsoid side code working and cleaned up, I'll move subpage handling support into this.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -116,48 +116,6 @@ var nameSpaceIds = {
...
- */
-
-/**
- * Checks whether namespace is `UserTalk`
- *
- * @method
- * @name Namespace.prototype#isUserTalk
- * @return {boolean}
- */
-
```
Comment (Pchelolo) [inline lib/index.js]: Docs in `README.md` are generated from these comments using the [jsdoc-to-markdown](https://www.npmjs.com/package/jsdoc-to-markdown) util. I think I need to add an npm script to generate API docs and run it as a pre-commit hook. So I think we shouldn't remove these comments.

File: lib/index.js [hunk 2]
```diff
@@ -178,6 +136,15 @@ Namespace.prototype.getNormalizedText = function() {
...
 
+/**
+ * Are subpages allowed for this namespace?
+ *
+ * @returns {boolean}
+ */
+Namespace.prototype.subpagesAllowed = function() {
+    return this._siteInfo.namespaces[this._id + ''].subpages !== undefined;
+};
+
```
Comment (Pchelolo) [inline lib/index.js]: In `RESTBase` to decrease the memory footprint of the `siteInfo` cache we delete all the properties that are not required for the module to work. A list of required properties is documented in the [SiteInfo](https://github.com/wikimedia/mediawiki-title/blob/master/lib/index.js#L16-L36) object documentation. Could you please add the `subpages` field there?

File: lib/index.js (hunk 1 above)
Comment (subbuss) [inline lib/index.js]: If you want to retain the comments, make sure to generate them for all 16 or so generated function. So, you prefer me to just get rid of this commit then?

File: lib/index.js (hunk 2 above)
Comment (subbuss) [inline lib/index.js]: Will do.

File: lib/index.js (hunk 1 above)
Comment (Pchelolo) [inline lib/index.js]: @subbuss Okey, let's get rid of them

**Pull request merged**